### DISCUSSION TITLE
useFormContext TypeError: cannot read property of null

### DISCUSSION THREAD

**Discussion by ghost** [react-hook-form/react-hook-form#9285] [Q&A]
I am moving components using React-Hook-Form from my main app to a reusable component library. When I move the components over, I get the error TypeError: cannot read property of null from useFormContext. I wanted to see if I could get the external library to work if I used the sample code provided on the React-Hook-Form website demonstrating useFormContext, which is similar in design to what I am using for my own code, but this got me the same error. Is it possible to useFormContext in a nested component if the component that is being imported from an external repo? Has anyone had any luck doing something similar?

I have seen a discussion on an issue similar to mine but the solution is now unavailable.

This is the code from the React-Hook-Form website, just with NestedInput being imported, if I can get this to work, I can resolve my specific issue.

import React from "react";
import { useForm, FormProvider } from "react-hook-form";
import { NestedComponent } from "external-library"

export default function App() {
  const methods = useForm();
  const onSubmit = data => console.log(data);

  return (
    <FormProvider {...methods} > // pass all methods into the context
      <form onSubmit={methods.handleSubmit(onSubmit)}>
        <NestedInput />
        <input type="submit" />
      </form>
    </FormProvider>
  );
}

import React from "react";
import { useFormContext } from "react-hook-form";

function NestedInput() {
  const { register } = useFormContext(); // retrieve all hook methods
  return <input {...register("test")} />;
}
export default NestedInput

**Comment by leapful** [comment]
@dalt12 Please provide a codesandbox url to re-produce the issue.
